pkg/cortexpb/timeseries.go

```diff
@@ -295,5 +297,6 @@ func ReuseTimeseries(ts *TimeSeries) {
 	}
 	ts.Labels = ts.Labels[:0]
 	ts.Samples = ts.Samples[:0]
+	ts.Exemplars = ts.Exemplars[:0]
```

Comment:
Look at the trick done above on labels to release pointers to the large gRPC buffer. We need to do the same for labels inside each exemplar.